controller/linodemachine_controller_helpers.go

```diff
@@ -198,21 +192,30 @@ func buildInstanceAddrs(ctx context.Context, machineScope *scope.MachineScope, i
 		Type:    clusterv1.MachineExternalIP,
 	})
 
-	// Iterate over interfaces in config and find VPC or VLAN specific ips
-	for _, iface := range configs[0].Interfaces {
-		if iface.VPCID != nil && iface.IPv4.VPC != "" {
-			ips = append(ips, clusterv1.MachineAddress{
-				Address: iface.IPv4.VPC,
-				Type:    clusterv1.MachineInternalIP,
-			})
-		}
-
-		if iface.Purpose == linodego.InterfacePurposeVLAN {
-			// vlan addresses have a /11 appended to them - we should strip it out.
-			ips = append(ips, clusterv1.MachineAddress{
-				Address: netip.MustParsePrefix(iface.IPAMAddress).Addr().String(),
-				Type:    clusterv1.MachineInternalIP,
-			})
+	// check if a node has vpc specific ip and store it
+	if len(addresses.IPv4.VPC) != 0 {
+		ips = append(ips, clusterv1.MachineAddress{
+			Address: *addresses.IPv4.VPC[0].Address,
+			Type:    clusterv1.MachineInternalIP,
+		})
+	}
```

Comment:
question: Just want to clarify, we are moving this logic out of the loop since we just have one VPC subnet?

Reply:
Previously, VPC ips were not returned in GetInstanceIPAddresses https://github.com/linode/linodego/blob/v1.30.0/instance_ips.go#L17-L22. This was implemented in newer versions and we now do get VPC specific ips in response of this call. This code is just to rely on that response. We now need ListInstanceConfigs only for VLAN specific ips as they are still not returned in GetInstanceIPAddresses call. The code restructures in a way that we make only one linode API call if we are not using VLANs.

Reply:
But your point is right, maybe I can add a for loop here to add all VPC ips? Currently we don't support more than one, so not sure if we really need it.

Reply:
If we can just note that as a comment for future reference that should be fine. We can add the loop in future if we support more than one VPC ip